### ISSUE TITLE
BiRNN bug

### ISSUE THREAD

**Issue by ruizhaogit** [tflearn/tflearn#208]
Hi TFlearn,

There is a small bug in BiRNN:

WARNING:tensorflow:Error encountered when serializing layer_tensor/BiRNN.
Type is unsupported, or the types of the items don't match field type in CollectionDef.
'list' object has no attribute 'name'.

Best,
Rui

**Comment by tgallice** [comment]
This issue come from the model serialization. On serialization, `tensorflow` try to get the `name` property for a `list` or `dict` value.

There are severals places where a `list` or `dict` is added as collection value :
- https://github.com/tflearn/tflearn/blob/master/tflearn/layers/recurrent.py#L92
- https://github.com/tflearn/tflearn/blob/master/tflearn/layers/recurrent.py#L400
- https://github.com/tflearn/tflearn/blob/master/tflearn/summaries.py#L51

[Here][1] `tensorflow` call the `name` property which is good for `Tensor` but not for `list` or `dict`. But the [_get_kind_type][2] does not make the distinction between `list or dict` and `Tensor`. 

We can maybe make a workaround with `tf.identity` ? @aymericdamien what do you think ? It's a misuse in `tflearn` or an issue for `tensorflow` ?

[1]: https://github.com/tensorflow/tensorflow/blob/73ced9d797056c7e67a06ed2098dd809d85ec44a/tensorflow/python/training/saver.py#L1217
[2]: https://github.com/tensorflow/tensorflow/blob/73ced9d797056c7e67a06ed2098dd809d85ec44a/tensorflow/python/training/saver.py#L1166-L1185

**Comment by aymericdamien** [comment]
Yes, the collections that have dicts (such as summaries) are deleted before being saved into the model file.
I think this is an error linked with tensorflow... before it was working well (you could save dicts without problems) but when they used protobuf, it got issues. But, I do not think TensorFlow will support dict any time soon, so we should find a way to fix that in TFLearn.

**Comment by ssriramana93** [comment]
Hello, it's been a year has this been fixed?

**Comment by saisumit** [comment]
has it been resolved ?
